File: Loenn/entities/winged_strawberry_direction_controller.lua
local sorbetHelper = require("mods").requireFromPlugin("libraries.sorbet_helper")

local wingedStrawberryDirectionController = {}

local directions = {
    "Up",
    "Down",
    "Left",
    "Right",
    "UpLeft",
    "UpRight",
    "DownLeft",
    "DownRight"
}

wingedStrawberryDirectionController.name = "SorbetHelper/WingedStrawberryDirectionController"
wingedStrawberryDirectionController.texture = "editorSprites/SorbetHelper/wingedStrawberryDirectionController"
wingedStrawberryDirectionController.depth = sorbetHelper.controllerDepth
wingedStrawberryDirectionController.placements = {
    name = "winged_strawberry_direction_controller",
    data = {
        direction = "Up"
    }
}

wingedStrawberryDirectionController.fieldOrder = {
    "x", "y",
    "direction"
}

wingedStrawberryDirectionController.fieldInformation = {
    direction = {
        options = directions,
        editable = false
    }
}

return wingedStrawberryDirectionController

File: Ahorn/entities/WingedStrawberryDirectionController.jl
module SorbetHelperWingedStrawberryDirectionController

using ..Ahorn, Maple

const directions = String[
    "Up",
    "Down",
    "Left",
    "Right",
    "UpLeft",
    "UpRight",
    "DownLeft",
    "DownRight"
]

@mapdef Entity "SorbetHelper/WingedStrawberryDirectionController" WingedStrawberryDirectionController(x::Integer, y::Integer,
    direction::String="Up")

const placements = Ahorn.PlacementDict(
    "Winged Strawberry Direction Controller (Sorbet Helper)" => Ahorn.EntityPlacement(
        WingedStrawberryDirectionController
    )
)

function Ahorn.editingOptions(entity::WingedStrawberryDirectionController)
    return Dict{String, Any}(
        "direction" => directions
    )
end

function Ahorn.selection(entity::WingedStrawberryDirectionController)
    x, y = Ahorn.position(entity)

    return Ahorn.Rectangle(x - 12, y - 12, 24, 24)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::WingedStrawberryDirectionController, room::Maple.Room) = Ahorn.drawImage(ctx, "objects/SorbetHelper/wingedStrawberryDirectionController/icon", -12, -12)

end
